utils/erasure.go

```diff
@@ -0,0 +1,38 @@
+package utils
```

Comment:
Why utils? 

I like the approach of making things reusable but in a modular domain-specific way, not a monolithic way. And we are already going with the former.

The latter:
services/
    header_service.go
    share_service.go

The former: 
header/
    service.go
share/
    service.go

Same with utils. Having one big folder for all the utils is monolithic and prone to dependency cycles, as it depends on our data structures.

If this makes sense to you, pls move you the helper func inside a respective module/domain package.